### DISCUSSION TITLE
How to avoid code splitting ?

### DISCUSSION THREAD

**Discussion by ahetawal-p** [parcel-bundler/parcel#8986] [Q&A]
Hello, 
I am creating a standalone single `.js` file using parcel for my react app. 
I am using this command to bundle all the code.
```
parcel build src/index.tsx --no-source-maps --dist-dir public/widget
```

The above works great, and I get a single output of index.ts in `public/widget` folder. 

Recently I added segment integration to my react app.
npm package :  `@segment/analytics-next`

After adding this I see multiple code split bundles are being created by parcel, and the output in the `public/widget` looks like this 
```
$ NODE_ENV=production yarn build:widget
$ parcel build src/index.tsx --no-source-maps --dist-dir public/widget
parcel-namer-rewrite: Rewrite index.js -> embed.js
✨ Built in 13.81s

public/widget/embed.js                            805.16 KB    10.39s
public/widget/auto-track.11b2606f.js                 1.8 KB     220ms
public/widget/query-string.47492dbd.js              1.33 KB     452ms
public/widget/routing-middleware.d4b8d49b.js       76.78 KB     516ms
public/widget/ajs-destination.a35b1512.js           10.3 KB     649ms
public/widget/legacy-video-plugins.9471845d.js        750 B     220ms
public/widget/index.umd.feb94f79.js                12.23 KB     452ms
public/widget/schema-filter.fd9812ff.js             1.74 KB     516ms
public/widget/remote-middleware.04b348ea.js         1.36 KB     649ms
```

But for the embed component to work, I would want to still have the single `embed.js` file.

How can achieve this ?

NOTE: Removing segment dependency and all related code, bring back the original intended bundle state.

**Comment by joshamaju** [comment]
Have you tried `includeNodeModules` field in targets?
